File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/state/TrackerScreenState.java
package ua.com.foxminded.locationtrackera.ui.tracker.state;

import ua.com.foxminded.locationtrackera.mvi.states.ScreenState;
import ua.com.foxminded.locationtrackera.ui.tracker.TrackerContract;

public class TrackerScreenState extends ScreenState<TrackerContract.View> {

    private final int gpsStatus;
    private final boolean serviceStatus;

    public TrackerScreenState(int gpsStatus, boolean serviceStatus) {
        this.gpsStatus = gpsStatus;
        this.serviceStatus = serviceStatus;
    }

    @Override
    public void visit(TrackerContract.View screen) {
        screen.changeGpsStatus(gpsStatus);
        screen.changeServiceStatus(serviceStatus);
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/state/TrackerScreenState.kt
package ua.com.foxminded.locationtrackera.ui.tracker.state

import ua.com.foxminded.locationtrackera.mvi.states.ScreenState
import ua.com.foxminded.locationtrackera.ui.tracker.TrackerContract

sealed class TrackerScreenState : ScreenState<TrackerContract.View>() {

    class ProvideStatus(
        private val gpsStatus: Int,
        private val serviceStatus: Boolean
    ) : TrackerScreenState() {
        override fun visit(screen: TrackerContract.View) {
            screen.changeGpsStatus(gpsStatus)
            screen.changeServiceStatus(serviceStatus)
        }
    }
}
